File: frontend/src/api/analysis.ts
import { apiClient } from '@/lib/api-client'
import type { AnalysisResponse, AnalysisStatusFilter, EvaluatedFilter } from '@/types/analysis'

export const analysisApi = {
  list: (params: {
    status?: AnalysisStatusFilter
    evaluated?: EvaluatedFilter
    page?: number
    per_page?: number
    project_id?: string
  }) =>
    apiClient.get<AnalysisResponse>('/analysis', { params }).then(r => r.data),
}

File: backend/api/analysis.py
from fastapi import APIRouter, Depends, Query

from backend.api.deps import get_analysis_service
from backend.models.analysis import AnalysisResponse
from backend.services.analysis import AnalysisService

router = APIRouter()


@router.get("", response_model=AnalysisResponse)
def list_analysis(
    status: str = Query("all", pattern="^(all|pending|approved|disapproved)$"),
    evaluated: str = Query("all", pattern="^(all|yes|no)$"),
    page: int = Query(1, ge=1),
    per_page: int = Query(20, ge=1, le=100),
    project_id: str | None = Query(None),
    svc: AnalysisService = Depends(get_analysis_service),
):
    return svc.get_analysis(
        status=status, evaluated=evaluated, page=page, per_page=per_page,
        project_id=project_id,
    )
